conference/user_panel.py

```diff
@@ -148,6 +137,29 @@ def privacy_settings(request):
     )
 
 
+@login_required
+def profile_settings(request):
+    attendee_profile = get_or_create_attendee_profile_for_new_user(user=request.user)
+
+    profile_form = ProfileSettingsForm(instance=attendee_profile)
+
+    if request.method == "POST":
+        profile_form = ProfileSettingsForm(
+            instance=attendee_profile, data=request.POST, files=request.FILES
+        )
+
+        if profile_form.is_valid():
+            profile_form.save()
+            # Read the saved data back to make sure things get saved correctly
+            profile_form = ProfileSettingsForm(instance=attendee_profile)
```

Comment:
how does that work?

Reply:
Well, I guess this is just a safety measure here; if the form only updates a single model, this line makes no sense. Here, since the form contains extra logic to read the data from associated models and then save it back, it adds extra points of failure - by reading it back the users get immediate feedback if something didn't work; otherwise, they'd get the data they've sent back - so if something failed to save/be read back, they'd be none the wiser.

It's a hack - when the fields can be moved to a single model, this won't be necessary.